commands/version_test.go

```diff
@@ -58,3 +61,386 @@ func Test_addVersion_short_version(t *testing.T) {
 		t.Fatalf("Output is not as expected:\n%s", stdOut)
 	}
 }
+
+func Test_gateway_uri(t *testing.T) {
```

Comment:
We have a lot of new tests here and repeated code. Do you think there's a good way to refactor to a set of test scenarios and sub-tests in a loop? @rgee0 is pretty good at this in alexellis/derek.

Reply:
@alexellis certainly could refactor the arrange part of the unit test, which would remove a lot of the duplication, i'll do that first and see how it looks

Reply:
okay i've taken a look at the `derek` repo, happy to refactor to that style

Reply:
As it stands I think the maintenance could be quite hard if we made some changes. You can also ping Richard for some guidance.